clients/python-wrapper/lakefs/client.py

```diff
@@ -118,15 +118,15 @@ def version(self) -> str:
 
 
 try:
-    DefaultClient = Client()
+    DEFAULT_CLIENT = Client()
 except NoAuthenticationFound:
     # must call init() explicitly
-    DefaultClient = None  # pylint: disable=C0103
+    DEFAULT_CLIENT = None  # pylint: disable=C0103
```

Comment:
I would suggest that we initialize the default client when it is first needed, rather than when loaded.  Rationales:

1. This swallows the exception.  So if I run without supplying a client and have no default authentication, I lose the authentication exception, and don't even have much information that I should have prepared it!
1. Another exception might be thrown here, and then loading my program fails.  But maybe my program doesn't even _need_ a default client!